### PR TITLE
[FEAT] Custom Sell Feat

### PR DESCRIPTION
**Pull request by TeissierYannis**
# Description

This feature includes a custom selling method. This allows users to enter the amount of plants that they want to sell.

Preview

## Type of change

- [X] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] This change requires a documentation update

# How Has This Been Tested?

I cannot try to sell in game because I need to have a test net farm. But I added test to sell.test.js, and I read my code a lot of time.

# Checklist:

- [X] Title of the PR is relevant and is prefixed with [FEAT], [CHORE] or [FIX]
- [X] Screenshot if it includes any UI changes
- [X] I have read the contributing guidelines and agree to the T&Cs
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/features/farming/shop/Plants.tsx [hunk 1]
```diff
@@ -21,17 +21,20 @@ export const Plants: React.FC = () => {
...
   const [isSellAllModalOpen, showSellAllModal] = React.useState(false);
+  const [isCustomSellModalOpen, showCustomSellModal] = React.useState(false);
   const { gameService } = useContext(Context);
   const [
     {
       context: { state },
     },
   ] = useActor(gameService);
   const [isPriceBoosted, setIsPriceBoosted] = useState(false);
+  const [amountToSell, setAmountToSell] = useState(0);
```
Comment (Pranav-yadav) [inline src/features/farming/shop/Plants.tsx:32]: are you sure type *number* is okay (or will be okay in future)  ?
Side Note: Inventory uses type *Decimal*.
Also have a look at `L:47`

File: src/features/farming/shop/Plants.tsx [hunk 2]
```diff
@@ -49,6 +52,11 @@ export const Plants: React.FC = () => {
     sell(1);
```
Comment (Pranav-yadav) [inline src/features/farming/shop/Plants.tsx:52]: the default param of `sell` is also set to `1`..

**Review by Pranav-yadav — CHANGES_REQUESTED** [review]

File: src/features/farming/shop/Plants.tsx [hunk 3]
```diff
@@ -49,6 +52,11 @@ export const Plants: React.FC = () => {
...
   };
 
+  const handleCustomSell = () => {
+    sell(amountToSell);
```
[the remark this answers is not in the document]
Comment (TeissierYannis) [reply src/features/farming/shop/Plants.tsx]: I made changes on that, is that okay ?

File: src/features/farming/shop/Plants.tsx [hunk 4]
```diff
@@ -119,6 +136,22 @@ export const Plants: React.FC = () => {
           >
             Sell All
           </Button>
```
[the remark this answers is not in the document]
Comment (TeissierYannis) [reply src/features/farming/shop/Plants.tsx:145]: I hide the Custom Sell button for value even at 5 at the moment

File: src/features/farming/shop/Plants.tsx (hunk 2 above)
Comment (TeissierYannis) [reply src/features/farming/shop/Plants.tsx]: I have removed the 1 to the "sell" function. The default param replace the empty value.

**Comment by Pranav-yadav** [comment]
I think we've already a open PR which covers all of these aspects...

**Comment by TeissierYannis** [comment]
> I think we've already an open PR #703 which covers all of these aspects...

Oh, mb, I don't see it.

**Comment by Rofrtd** [comment]
Hi @TeissierYannis, thanks for your contribution, but we do have another open PR #703 that is addressing this.
You can input your opinions on that PR if you wish. 
I'll close this PR for the time being. 
Thank you.

**Pull request closed without merging**